### DISCUSSION TITLE
Enforce unique hostname

### DISCUSSION THREAD

**Discussion by domainzero** [netbox-community/netbox#10122] [Q&A] [answered]
Is there a way to enforce uniqueness via custom validator? I found the unique function being used for `asset_tag`, but am not sure how to apply that to a different field like `name`.

Is this feasible?

**Comment by jeremystretch** [comment, marked as answer]
Yes; you can write a [custom validator class](https://docs.netbox.dev/en/stable/customization/custom-validation/#custom-validation-logic) to enforce whatever logic you'd like.

**Reply by domainzero** [reply]
Is there a `unique` function I can call for this, or do I have to build my own logic?

**Reply by domainzero** [reply]
Ended up figuring it out. Here's the code I used.
  

```python
from extras.validators import CustomValidator
from dcim.models import Device

class DeviceNameUnique(CustomValidator):
    def validate(self, device):
        existing_devices = Device.objects.filter(name__iexact=device.name).exists()
        print(existing_devices)
        if existing_devices:
            self.fail("Device name already in use.", field="name")
```

**Reply by LHBL2003** [reply]
It is not clear to me where I have to write this validation class.
Can you tell me which file I need to write the code to?

**Reply by LHBL2003** [reply]
So I managed to do it for Host/Devices and Virtual Machine and documented it here. What you had to enter where.:
https://github.com/netbox-community/netbox/discussions/16267
